File: nivel-04/ex-04/indice.js
var assert = require('assert');

var vet    = [6, 10, 4, 21, 9],
    iMaior = 0,
    iMenor = 0,
    i = 0;

for(i = 0; i < vet.length; i++) {
    if (vet[i] > vet[iMaior]) {
        iMaior = i;
    }

    if (vet[i] < vet[iMenor]) {
       iMenor = i;
    }
}

//
// Seu teste
//
try {

    assert.equal(3, iMaior);
    assert.equal(2, iMenor);

} catch(e) {
    console.log(e);
}

File: nivel-04/ex-04/indice.py
# Retornar os índices do maior e menor elemento 
# em uma lista usando laço de repetição

lista = [4, 9, 2, 11, 32, 7]
i_maior = 0
i_menor = 0

i = 0 
while i < len(lista):
	if lista[i] >= lista[i_maior]:
		i_maior = i

	if lista[i] <= lista[i_menor]:
		i_menor = i

	i += 1

assert i_maior == 4 , 'Maior elemento é o de índice = 4'
assert i_menor == 2 , 'Menor elemento possui índice = 2'